core/src/main/java/org/axonframework/eventsourcing/AggregateSnapshotter.java

```diff
@@ -86,9 +89,30 @@ public AggregateSnapshotter(EventStore eventStore, List<AggregateFactory<?>> agg
      */
     public AggregateSnapshotter(EventStore eventStore, List<AggregateFactory<?>> aggregateFactories,
                                 ParameterResolverFactory parameterResolverFactory) {
+        this(eventStore,
+             aggregateFactories,
+             parameterResolverFactory,
+             new ClasspathHandlerDefinition(Thread.currentThread().getContextClassLoader()));
```

Comment:
Instead of using the context class loader, use the class loader of the first aggregate factory provided.